src/badguy/badguy.cpp

```diff
@@ -325,7 +382,10 @@ void
 BadGuy::active_update(float dt_sec)
 {
   if (!is_grabbed())
-    m_col.set_movement(m_physic.get_movement(dt_sec));
+    m_col.set_movement(m_physic.get_movement(dt_sec) * (m_water_affected ? (
+      Vector((is_in_water() && !m_frozen) ? 0.7f : (is_in_water() && m_frozen) ? 0.1f : 1.f,
+             (is_in_water() && !m_frozen) ? 0.3f : (is_in_water() && m_frozen) ? 0.6f : 1.f)) :
+      Vector(1.f, 1.f)));
```

Comment:
Please split these ternary operators into separate if statements.

Reply:
Why

Reply:
Nested ternary operators are generally pretty unpleasant to read. They jam everything into a single line instead of going step by step. Ternary operators should only be used for simple cases.

Reply:
I saw this on the GitHub channel on Discord and thought "this cant be *that* bad".

It is that bad.

I cant read this lol.

Reply:
Alright, I can fix it